### PR TITLE
Issue redeem calculations

### PR DESCRIPTION
**Pull request by monsieurleberre**
A component to calculate quantities involved in issuance and redemption of compositions.
Also removed the `Weight` from `IComponentQuantity`, as it doesn't make sense outside a valuation context.

### CODE REVIEW

File: src/Trakx.Common/Core/ComponentQuantity.cs [hunk 1]
```diff
@@ -12,25 +12,23 @@ public class ComponentQuantity : IComponentQuantity
...
                 0, 18);
 
             ComponentDefinition = definition;
             Quantity = unscaledQuantity.ScaleComponentQuantity(ComponentDefinition.Decimals, indiceNaturalUnit);
-            Weight = weight;
 
             Debug.Assert(this.IsValid());
         }
 
+        public ComponentQuantity(IComponentQuantity quantity, decimal? newScaledQuantity = default)
```
Comment (AymericNoel) [inline src/Trakx.Common/Core/ComponentQuantity.cs:26]: Is it a constructor that takes a scaled quantity ? It's quite practical

File: src/Trakx.Drafts/Tools/EthereumBalanceViewer.cs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        public EthereumBalanceViewer(ITestOutputHelper output)
+        {
+            _output = output;
+            _flurlClient = new FlurlClient(@"https://api.ethplorer.io");
+        }
+
+        [Fact]
+        public async Task GetEthereumBalances()
+        {
+            var address = "0x9DC57560Dc50cCFB7475B78B594B2444CFB7a1A5";
```
Comment (AymericNoel) [inline src/Trakx.Drafts/Tools/EthereumBalanceViewer.cs:30]: Is it the Trakx address for which we're looking for tokens balances ?

**Review by AymericNoel — APPROVED** [review]
I've seen that you have added a `IssueRedeemCalculator` but I didn't see where you use it ? 
Is that means that we're going to issue and redeem for real soon ?
The tokens balance retriever is for displaying on the frontend the Trakx balance ?

File: src/Trakx.Drafts/Tools/EthereumBalanceViewer.cs (hunk 2 above)
Comment (monsieurleberre) [reply src/Trakx.Drafts/Tools/EthereumBalanceViewer.cs:30]: yes this is the address at which we currently have way too much 😨

File: src/Trakx.Common/Core/ComponentQuantity.cs (hunk 1 above)
Comment (monsieurleberre) [reply src/Trakx.Common/Core/ComponentQuantity.cs:26]: yes then you can get an almost clone of the quantity but with changed quantity :)

**Pull request merged**